It seems that `vcpkg_cmake_configure` does not change the cwd in contrast to `vcpkg_configure_cmake` on windows. 
The line `file(MAKE_DIRECTORY generated)` generates the following error on windows but not on other platforms:
```
CMake Error at CMakeLists.txt:13 (file):
  file attempted to create a directory:
  C:/Users/LeanderSchulten/git_projects/vcpkg/buildtrees/offscale-libetcd-cpp/src/fdb87cf638-60a9ea69fd.clean/generated
  into a source directory.
```

--- a/ports/offscale-libetcd-cpp/install-debug.patch
+++ b/ports/offscale-libetcd-cpp/install-debug.patch
@@ -1,0 +1,11 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 0d9aad9..7bdde1b 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -77,5 +77,5 @@ install(TARGETS etcdcpp EXPORT etcdcpp DESTINATION lib)
+ 
+ if (${CMAKE_BUILD_TYPE} STREQUAL "Release")
+ 	install(FILES ${proto_hs} DESTINATION include/libetcd/)
+-	install(EXPORT etcdcpp DESTINATION share/etcdcpp/ FILE etcdcppConfig.cmake)
+ endif()
++install(EXPORT etcdcpp DESTINATION share/etcdcpp/ FILE etcdcppConfig.cmake)

--- a/ports/offscale-libetcd-cpp/portfile.cmake
+++ b/ports/offscale-libetcd-cpp/portfile.cmake
@@ -4,15 +4,17 @@ vcpkg_from_github(
     REF 8607d8d7080c6eb447bc41b799a24bfdb87cf638
     SHA512 9bf4bf14fd52f4f6bbf8cf68de61e6a19eeffbdc501f05c8f614b5f13f206410884afd7fe226a077448e58e02990c65a6b27ee895ed34ba5ee701abe0500b875
     HEAD_REF master
-    PATCHES fix-dependency-grpc.patch
+    PATCHES
+        fix-dependency-grpc.patch
+        install-debug.patch
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    DISABLE_PARALLEL_CONFIGURE # see https://github.com/microsoft/vcpkg/pull/21168#issuecomment-961588989 why
 )
-vcpkg_install_cmake()
-
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME etcdcpp)
 vcpkg_copy_pdbs()
 
-file(INSTALL ${SOURCE_PATH}/LICENSE-MIT DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE-MIT" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/offscale-libetcd-cpp/vcpkg.json
+++ b/ports/offscale-libetcd-cpp/vcpkg.json
@@ -1,11 +1,19 @@
 {
   "name": "offscale-libetcd-cpp",
-  "version-string": "2019-07-10",
-  "port-version": 2,
+  "version-date": "2019-07-10",
+  "port-version": 3,
   "description": "A C++ client library for etcd. etcd is a distributed, reliable key-value store.",
   "homepage": "https://github.com/offscale/libetcd-cpp",
   "dependencies": [
     "grpc",
-    "protobuf"
+    "protobuf",
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
   ]
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4782,7 +4782,7 @@
     },
     "offscale-libetcd-cpp": {
       "baseline": "2019-07-10",
-      "port-version": 2
+      "port-version": 3
     },
     "ogdf": {
       "baseline": "2019-08-23",

--- a/versions/o-/offscale-libetcd-cpp.json
+++ b/versions/o-/offscale-libetcd-cpp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "70eb4516cd664f01e89d2701ae45ba05ea4c9d7a",
+      "version-date": "2019-07-10",
+      "port-version": 3
+    },
+    {
       "git-tree": "58aea86d7da68c1cad8d684197c39e23945b8d91",
       "version-string": "2019-07-10",
       "port-version": 2

